### ISSUE TITLE
Users with read permission should be able to see more details (Side-components) [5]

### ISSUE THREAD

**Issue by SahaLinaPrueger** [hzi-braunschweig/SORMAS-Project#8463]
### Problem Description

Sormas2Sormas gives the possibility that you can share a case/contact without hand over the ownership. So the user from target system can read/ view the entries, but can not edit them. Sormas2Sormas gives also the possibility to share a case/contact with hand over the ownership. So the user from source system can read/ view the entries, but can not edit them. In the following this is described as "read permission".
In Sormas you also often have the possibility to click on a "pencil-button". If you click this button a pop-up windows opens and you can see more details. 
In the future it should be possible for a user with read permission to click on the "pencil-button" to see more details. 

Currently it is not possible for a user, who only have read permission, to click on the "pencil-button" and see more details described in the pop-up-window. This means that a lot of information cannot be retrieved and is therefore lost.

### Proposed Change

A user with read permission can see more details by clicking on the "pencil-button".
On the following places, the "pencil-button" should be possible to click: 
_please notice that it should NOT be possible to edit for example new vaccinations with read permission, only the 'pencil button' should be clickable, so the entries can be opened and seen but can NOT be edited_

1. In a case:
      tab _case_ on the right side:
1.0 pencil button under _+new task_ (special: yet made tasks should still be editable)
1.1 pencil button under +new sample (+ click on the lab message should open the lab message)
1.2 pencil button under +link event
1.3 pencil button under +new vaccination
1.4 pencil button under +new travel entry
1.5 pencil button under +new report
1.6 _create_ button should be clickable and it should be possible to create a document

      tab _case person_:
1.7 pencil button under _Addresses_
1.8 pencil button under Person _contact details_

      tab _hospitalization_:
1.9 pencil button under _Previous Hospitalization_

      tab _epidemiological data_
1.10 pencil button under _Exposure Investigation_
1.11 pencil button under _Activity as Case_
1.12 pencil button under _+ new contact_
1.13 pencil button under _+ new travel entry_

      tab _therapy_
1.14 pencil button under _Prescriptions_
1.15 pencil button under _Executed treatments_

      tab _follow-up_
1.16 pencil button under _+new visit_

      tab _clinical course_
1.17  pencil button under _clinical assessments/ new clinical assessments_

2. sample
2.1 pencil button under +new test result (pathogen tests)
2.2 pencil button under +new test result (additional tests)

3. contact
      tab _contact_ on the right side:
3.0 pencil button under _+new task_ (special: yet made tasks should still be editable)
3.1 pencil button under +new sample (+ click on the lab message should open the lab message)
3.2 pencil button under +link event
3.3 pencil button under +new vaccination
3.4 pencil button under +new travel entry
3.5 pencil button under +new report
3.6 _create_ button should be clickable and it should be possible to create a document

      tab _contact person_:
3.7 pencil button under _Addresses_
3.8 pencil button under Person _contact details_

      tab _epidemiological data_
3.9 pencil button under Exposure Investigation

      tab _follow-up visits_
3.10 pencil button under +new visit

### Implementation Details

- Entry point: `LayoutWithSidePanel`
-  replace pencil with eye icon and hide create button

### Additional Information

![image](https://user-images.githubusercontent.com/100844689/159257339-a94590a0-f6c9-4df8-87cc-600bb9956524.png)
![image](https://user-images.githubusercontent.com/100844689/159257419-b277234c-29ae-40f2-96a3-5ad9b004e165.png)

**Comment by leventegal-she** [comment]
@SahaLinaPrueger 

Allowing access to click on pencil buttons was not asked until now so there is only a simple readonly view where everithing is disabled and nothing is clickable.

It is not necessarily linked to S2S, it is general for readonly view, e.g when opening an archived case.

I don't consider this to be a bug. It is a new feature that should be refined before starting to implement it.
There are some places where it is relatively easy to do like the pencil buttons on the boxes at the right side of the page, but others like person addresses or contact details would need more work to be implemented.

@MateStrysewske @MartinWahnschaffe @markusmann-vg

**Comment by dragosb-vg** [comment]
Reopened the issue as the "Tasks" and "Users" menus can no longer be accessed on live instances: 1.76.0-SNAPSHOT [6a8a0a2](https://github.com/hzi-braunschweig/SORMAS-Project/commits/6a8a0a2521dda3cebd536e08f0fae0c07ea0c229)

**Comment by roxanamlendea** [comment]
I'm reopening the ticket because of a new finding: the Report card is not read-only for users with read permission only.

![image](https://user-images.githubusercontent.com/89522688/201357599-33482afa-4622-41dd-9343-ccec8e0440f3.png)

**Comment by roxanamlendea** [comment]
Another finding is that the Prescriptions are no longer editable:

1. Log in with an Admin + National User;
2. Open a Case and navigate to the Therapy tab;
3. Create a Prescription;
4. Then open the Prescription from step 3;
-> the Prescription is read-only, but it should be editable;

**Comment by roxanamlendea** [comment]
I'm documenting here 2 further findings:

A) In read-only mode, on the Therapy, Follow-up and Clinical Assessment tabs, the rows of the last entries that were opened remain highlighted, even after the pop-up window is closed.

![image](https://user-images.githubusercontent.com/89522688/202177131-2f7993b5-b580-49f1-9e87-9c59edf1c6a5.png)


B) For soft deleted Cases and Contacts, the side components are active, they are not read-only:
1. Open a Case;
2. Copy the URL;
3. Soft delete the Case;
4. Paste the URL;
-> the side components are no longer read-only;

![image](https://user-images.githubusercontent.com/89522688/202177951-ad8772e4-ce60-4277-9039-a64964ccf0a1.png)

**Comment by roxanamlendea** [comment]
The implementation needs to take into consideration also the User Rights - the users that have CASE_VIEW = YES and CASE_EDIT = NO (eg. National Observer, Region Observer etc.);
Currently, these users are able to open only some of the side components in read-only mode:
![image](https://user-images.githubusercontent.com/89522688/202214282-e86cb4f9-e637-4efa-b97a-0df67a321be3.png)
![image](https://user-images.githubusercontent.com/89522688/202214346-fecda2fc-2d3e-42a4-9183-aca0104e4e37.png)

Also, the entries on the other tabs cannot be opened in read-mode because the eye icon is missing:
![image](https://user-images.githubusercontent.com/89522688/202214566-577070de-e3dc-4d46-aace-bbe04141cd53.png)

**Comment by sergiupacurariu** [comment]
I attached a description of the behavior implemented: 
[# 8463 - implemented behavior explanation.xlsx](https://github.com/hzi-braunschweig/SORMAS-Project/files/10081711/8463.-.implemented.behavior.explanation.xlsx)

**Comment by SahaLinaPrueger** [comment]
@roxanamlendea 
tested this for cases on nightly test-de1

1. eye button for _vaccinations_  and _reports_ are not clickable -> entry does not open
2. only a design thing: letters are _not_ greyed out on some places, could be confusing for the users. On 1.73-1.76. the letters were greyed out everywhere. I guess it is not possible to grey them out, because the _eye button_ still needs to be clickable, right? But if it is possible to handle it like on the tab _case_, where the letters of the boxes are greyed out but the eye button is still clickable, this would be nice.
3. please check this points for contacts, too.
![Hotfix 1 77 n](https://user-images.githubusercontent.com/100844689/203823024-8f204b16-52c1-4711-b2c7-19353f6c9a7b.gif)

**Comment by SahaLinaPrueger** [comment]
For the document-thing I opened a new ticket: #11018

**Comment by roxanamlendea** [comment]
@SahaLinaPrueger , sorry for answering this late but I was off for the last days.
1) I'm reopening the ticket for the issue regarding the eye button for vaccinations and reports that are not clickable.
2) We are aware of this and it will be treated in a separate issue. A new ticket will be logged for this.
3) The Contacts tab wasn't included in the description of the ticket, so the implementation was not done for this tab. I noticed this while testing and I opened a change request for this: #10979

**Comment by SahaLinaPrueger** [comment]
@roxanamlendea thank you very much.
FYI: Point tree is not about the _tab contacts_ . 
This _tab contacts_ is _not_ included in the ticket, because this tab should be clickable. In one Version this wasn't possible and therefore I created a bug: #9838 If this is not possible on 1.77 this is a finding, too. (on 1.76 it is possible). 
Should i open a new ticket for this?

With point three i meant, that i did not testes this issue for shared contacts and maybe also for contacts the "eye button" on reports and vaccinations is not clickable. But i had no time to check this yesterday, therefore i wrote this. that was probably unfortunate expressed.

**Comment by roxanamlendea** [comment]
Ticket was verified on local setup using version of Sormas 1.78.0-SNAPSHOT in the week 5-9 dec, I just forgot to write the comment then.

**Comment by adinaflorea9** [comment]
@SORMAS-SahaLinaPrueger I tagged you here as I have a short question but I do not have another possibility of messaging you. 

This is related to tasks of entities that have been shared with handover ownership from the source system into the target system. Is it the correct and expected behavior that in the source system those tasks are still editable completely even though the entity is only 'To view' in the source system now?

I am asking because of the following scenario that 'alters' the information that was synced on the case from the target system back in the source system:

1. In the source system create a case that has a 'Case investigation task';
2. Share the case with handover ownership into a target system;
3. Accept the case in the target system;
4. Open the case in the target system and change the 'Investigation status' to 'Done';
5. Open the case in the source system and refresh the page so that the sync is done 
-> the 'Investigation status' was now updated as per the sync of the information from the target system.
6. In the source system, from the case page open the existing task and update the 'Task Status' to 'Removed';
7. Click 'Save' in the task form;
-> on the case page, the 'Investigation status' is now 'Investigation discarded' and no longer the information that was synced from the target system. Should this be possible? Which should take prevalence - the information from the target system that should be synced or the editing of that task after the case was already handed over from the source system and is now only to view, and not editable?

**Comment by SORMAS-SahaLinaPrueger** [comment]
@adinaflorea9 thank you for the question. 

The desired behaviour in the last year was as follows:
1. Tasks should not be shared in general, because tasks and the work methods behind them and the names of health department employees only concern the source system. Each health department has its own workflow and a different number of responsible teams. "Case investigation task" can mean that the case has been contacted and appointments for sample collection have been made in one health department and in another department it can mean that the case has only been contacted. For employees in the target system, it would be confusing to see the tasks from the source system, as they would not be able to distinguish at first glance whether the task is one from the source system or from their target system. Handing over the tasks could therefore have serious consequences and disrupt the workflow because staff in the target system would assume that a task has already been done, even though this task has not yet been done in the target system. In addition, as always, data protection plays a role. 

The S2S manual therefore states:  
"When sharing with the ownership, all information available in SORMAS for the corresponding entities (case or contact) is transferred except for the tabs:
Follow-up, as well as the entities Tasks, Events, Reports, Documents and Document Templates, and the fields General Comment, External ID, External File Number and internal file number." 
-> Notice: Reports could not yet be shared at that time. 

If data protection issues has been solved and the bavarian people wish that the tasks should be shared my answer would be:
Once the tasks where shared then there should not be any sync in any direction.

If there has to be a sync, then my answer would be:
Sync should only be the direction from target system to source system.
and if the situation you described is ongoing the source system has prevalance. The reason why the tasks are still editable even if the share was handed over with ownership is that some health departments use the tasks to start and finish their symptom diary via C.I.S.S. So they have to change the task status and then the symptom diary ends. If they have forgot to do it before sharing then they can do it afterwards.
And the offices need to be able to complete their workflows in an orderly manner even after the case has been handed over. So if a team is responsible for closing certain electronic files, they need to be able to mark this task as "done" even after sharing with ownership. 

I hope the answer is helpful - if not please feel free to ask anytime

**Comment by adinaflorea9** [comment]
@SORMAS-SahaLinaPrueger Thank you for your answer and for explaining this in such detail!
